SKA-ScienceDataProcessor/integration-prototype
# TSK-2753 Updates to the EC Master Controller

## Description:
Slight refactor of Master Controller to emulate subsystems changing target state to current state.

TSK 2753: Restructure of the Master Controller Application

## Testing instructions:
Using the Docker stack deployed for the 23/11/2018 demonstration, remove the sip_ec_master_controller docker container and run the master controller from its location in the sandbox.

_All command described below are assumed to be run from the top level SIP repo directory._

1. Create a Python Virtual environment and activate it:
```bash
python3 -m venv venv
source venv/bin/activate
```

2. Install python packages into virtual environment from the requirements.txt file:
```bash
pip install -r sip/execution_control/master_controller/requirements.txt
```

3. In another terminal, run the Docker stack and delete the existing master controller container:
```bash
docker stack deploy -c sip/execution_control/master_controller/docker-compose.dev.yml mc_test
```

4. From original terminal run the master controller:
```bash
cd sip/execution_control/master_controller
python3 -m app -v
```

5. Get an iTango prompt
```bash
./tools/get_itango_prompt.sh
```

6. Set the state of the SDP to 'on' by getting a client to the Tango Master and setting the target state of SDP to on.
```python
md = DeviceProxy('sip_sdp/elt/master')
md.set_target_state('on')
```

If this command works correctly this should be reflected in the Master Controller log which will report
```bash
INFO - SDP target state changed to 'on'
```

## Types of changes

## Checklist:
